budget-frontend/src/components/Register/Register.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import {
+    Button,
+    Form,
+    Input,
+} from 'antd';
+
+interface DataNodeType {
+    value: string;
+    label: string;
+    children?: DataNodeType[];
+}
+
+const formItemLayout = {
+    labelCol: {
+        xs: { span: 24 },
+        sm: { span: 8 },
+    },
+    wrapperCol: {
+        xs: { span: 24 },
+        sm: { span: 16 },
+    },
+};
+
+const tailFormItemLayout = {
+    wrapperCol: {
+        xs: { span: 24, offset: 0 },
+        sm: { span: 16, offset: 4 },
+        lg: { span: 16, offset: 4 },
+    },
+};
+
+export const Register: React.FC = () => {
+    const [form] = Form.useForm();
+
+    const onFinish = (values: any) => {
+        console.log('Received values of form: ', values);
+    };
+
+    return (
+        <Form
+            {...formItemLayout}
+            form={form}
+            name="register"
+            onFinish={onFinish}
+            style={{ minWidth: 500 }}
+            scrollToFirstError
+        >
+            <Form.Item
+                name="email"
+                label="E-mail"
+                rules={[
+                    {
+                        type: 'email',
+                        message: 'The input is not valid E-mail!',
+                    },
+                    {
+                        required: true,
+                        message: 'Please input your E-mail!',
+                    },
+                ]}
+            >
+                <Input />
+            </Form.Item>
+
+            <Form.Item
+                name="password"
+                label="Password"
+                rules={[
+                    {
+                        required: true,
+                        message: 'Please input your password!',
+                    },
+                ]}
+                hasFeedback
+            >
+                <Input.Password />
+            </Form.Item>
+
+            <Form.Item
+                name="confirm"
+                label="Confirm Password"
+                dependencies={['password']}
+                hasFeedback
+                rules={[
+                    {
+                        required: true,
+                        message: 'Please confirm your password!',
+                    },
+                    ({ getFieldValue }) => ({
+                        validator(_, value) {
+                            if (!value || getFieldValue('password') === value) {
+                                return Promise.resolve();
+                            }
+                            return Promise.reject(new Error('The new password that you entered do not match!'));
+                        },
+                    }),
+                ]}
+            >
+                <Input.Password />
```

Comment:
will be good to provide some accessibility enhancements for the inputs:
example:
```js
<Input aria-label="Email input" />
```